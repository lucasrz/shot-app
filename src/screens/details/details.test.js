import React from 'react';
import renderer from 'react-test-renderer';
import Details from './index';
import store from '../../store';

describe('Details Component', () => {
  it('Should insttiate the component', () => {
    const match = {
      params: {
        id: 1 
      }
    }
    
    const history = {
      push: jest.fn()
    }
    const tree = renderer
    .create(<Details store={store} match={match} history={history}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  
  it('Testing Details Component when receiving no id', () => {
    const match = {
      params: {
        id: null 
      }
    }
    
    const history = {
      push: jest.fn()
    }
    
    const tree = renderer
    .create(<Details store={store} match={match} history={history}/>)
    .toJSON();
    expect(history.push).toHaveBeenCalledWith('/');
  });
});