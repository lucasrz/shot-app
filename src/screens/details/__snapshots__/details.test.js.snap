// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component Should insttiate the component 1`] = `
<div
  className="container"
>
  <h1>
    Loading...
  </h1>
</div>
`;
